app.rb

```diff
@@ -0,0 +1,180 @@
+require './book'
+require './person'
+require './decorator'
+require './rental'
+require './student'
+require './teacher'
+require './classroom'
+require_relative 'nameable'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def run
+    puts 'Welcome to School Library App!'
+    loop do
+      interface_menu
+      option = gets.chomp
+      if option == '7'
+        puts 'Thank you for using the Library App!'
+        break
+      end
+      option_selector(option)
+    end
+  end
+
+  def list_books
+    @books.each { |book| puts "#{book.title} by #{book.author}" }
+  end
+
+  def list_people
+    @people.each { |person| puts "#{person.name} - #{person.age}" }
+  end
+
+  def list_rentals_person
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index { |person, index| puts "#{index}) #{person.name}" }
+    person_index = gets.chomp.to_i
+    show_rentals_person(person_index)
+  end
+
+  def show_rentals_person(person_index)
+    puts "#{@people[person_index].name} has rented the following books:"
+    @people[person_index].rentals.map do |rental|
+      puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}"
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]'
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'That is not a valid input'
+      nil
+    end
+  end
+
+  def create_student
+    id = @people.length + 1
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Has parent permission? [Y/N]?'
+    permission = gets.chomp
+    if permission.match?(/y/i)
+      parent_permission = true
```

Comment:
- [ ] When creating a student, as soon as I enter any value for parent permission, the app crashes with an error saying a wrong number of arguments. Kindly refer to the screenshot below:

<img width="1230" alt="Screenshot 2023-05-27 at 8 07 35 PM" src="https://github.com/Odongo006/OOP-School-Library/assets/50941074/b2cb178f-e1ac-47a1-9e95-33a075a27862">